go/vm/lfvm/interpreter_test.go

```diff
@@ -69,24 +69,39 @@ func getContext(code Code, data []byte, runContext vm.RunContext, stackPtr int,
 
 // Test UseGas function and correct status after running out of gas
 func TestGasFunc(t *testing.T) {
-	ctx := getEmptyContext()
-	ctx.gas = 100
-	ok := ctx.UseGas(10)
-	if !ok {
-		t.Errorf("expected not failed useGas function, got failed")
-	}
-	if ctx.gas != 90 {
-		t.Errorf("expected gas in context is 90, got %d", ctx.gas)
-	}
-	ok = ctx.UseGas(100)
-	if ok {
-		t.Errorf("expected failed useGas function, got ok")
-	}
-	if ctx.gas != 90 {
-		t.Errorf("expected gas in context is 90 also after failing, got %d", ctx.gas)
+
+	tests := []struct {
+		name      string
+		gas       vm.Gas
+		amount    vm.Gas
+		wantedGas vm.Gas
+		expected  bool
+		status    Status
+	}{
+		{"Zero amount", 100, 0, 100, true, RUNNING},
+		{"Sufficient gas", 100, 10, 90, true, RUNNING},
+		{"Insufficient gas", 10, 100, 10, false, OUT_OF_GAS},
+		{"All gas", 100, 100, 0, true, RUNNING},
+		{"Negative amount", 100, -100, 200, true, RUNNING},
```

Comment:
Not sure negative gas should be supported in a function that is named "UseGas". 

The problem is that such a behavior could potentially lead to non-termination of the EVM, if for some reason a bug produces some negative gas costs. The gas pool would no longer get strictly reduced.

I would suggest that a negative gas value is handled like a infinite gas cost. Wdyt?

Reply:
It is early for me to do these judgements, but I begin to think that negative gas values, when dealing with costs, are symptoms of overflows. 
I see two options, treat it as infinite gas, or introduce overflow checks on every gas computation. 

- The first assumes that a negative gas value is indeed an overflow.
- The latter quite a big change. 

I cannot tell you at this time how severe is this.

Reply:
UseGas  shall no longer accept negative costs, by returning out of gas... unit testing and ct pass. This unblocks fuzzing.